doc/source/dev/contributor/meson_advanced.rst

```diff
@@ -0,0 +1,52 @@
+.. _meson-advanced:
+
+Build with Meson Optimizations
+==============================
+
+Meson provides different optimizations levels while configuring the project. You can see
+the available options for optimizations at
+`meson documentation <https://mesonbuild.com/Builtin-options.html#core-options>`.
+
+Assuming that you are building from scratch(do ``git clean -xdf`` if needed), you can
+configure the build as following::
+
+    meson setup build --optimization g --prefix=$PWD/build-install
+
+Now, you can use the ``dev.py`` interface for further building, installing and testing SciPy::
+
+    python dev.py -s linalg
+
+This will work because after initial configuration, Meson will remember the config options.
+
+
+Use GCC and Clang build in parallel
+===================================
+
+As discussed, Meson is fully out-of-place, so different builds will not interfere
+with each other. We assume in the rest of this section that GCC is the default.
+For example, let us build using GCC and Clang.
+
+1. Build with GCC::
+
+    python dev.py --only-build
+
+Using the above command, meson will build with gcc compiler in ``build`` directory.
+It will then install SciPy into ``$PWD/build-install/lib/python3.x/site-packages/scipy``.
+
+2. Build with Clang::
+
+    CC=clang CXX=clang++ python dev.py --build-only --build-dir=build-clang
```

Comment:
Would be nice to also show the magic env var for specifying the Fortran compiler, while you're at it.

Reply:
Something like this?
```
CC=clang CXX=clang++ FC=gfortran python dev.py --build-only --build-dir=build-clang
```

Reply:
yep